tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Builders/OrderSummaryDataBuilder.cs

```diff
@@ -0,0 +1,119 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using NHSD.BuyingCatalouge.Ordering.Api.Testing.Data.Entities;
+using NHSD.BuyingCatalouge.Ordering.Api.Testing.Data.EntityBuilder;
+using NHSD.BuyingCatalouge.Ordering.Api.Testing.Data.Data;
+
+namespace NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests.Builders
+{
+    internal sealed class OrderSummaryDataBuilder
+    {
+        private OrderEntity _orderEntity;
+        private ServiceRecipientEntity _serviceRecipientEntity;
+        private OrderItemEntity _catalogueSolutionEntity;
+        private OrderItemEntity _associatedServiceEntity;
+        private OrderItemEntity _additionalServiceEntity;
+
+        private OrderSummaryDataBuilder(string orderId="C000016-01")
+        {
+            WithOrderEntity(
+                OrderEntityBuilder.Create()
+                    .WithOrderStatusId((int)OrderStatus.Unsubmitted)
+                    .WithOrderId(orderId)
+                    .WithDescription("A Description")
+                    .WithOrganisationId(new Guid("4af62b99-638c-4247-875e-965239cd0c48"))
+                    .WithServiceRecipientsViewed(true)
+                    .WithAdditionalServicesViewed(true)
+                    .WithCatalogueSolutionsViewed(true)
+                    .WithAssociatedServicesViewed(true)
+                    .WithFundingSourceOnlyGMS(true)
+                    .Build());
+        }
+
+        public static OrderSummaryDataBuilder Create(string orderId="C000016-01") =>
+            new OrderSummaryDataBuilder(orderId);
+
+        public OrderSummaryDataBuilder WithOrderEntity(OrderEntity orderEntity)
+        {
+            _orderEntity = orderEntity;
+            return this;
+        }
+
+        public OrderSummaryDataBuilder WithServiceRecipientViewed(bool viewed)
+        {
+            if (_orderEntity != null)
+            {
+                _orderEntity.ServiceRecipientsViewed = viewed;
```

Comment:
The idea behind this builder pattern is delayed creation. By applying this change here it relies on the client of this class to know the order to call the methods in. I would like to suggest the adding of a variable e.g. _serviceRecipientsViewed, which then gets applied to the order in the `Build` method. Obviously this applies to the other "viewed" properties. What do you think?